devito/pointdata.py

```diff
@@ -10,7 +10,23 @@
 __all__ = ['PointData']
 
 
-class PointData(DenseData):
+class CompositeData(DenseData):
+    """
+    Base class for DenseData classes that have DenseData children
+    """
+
+    is_CompositeData = True
+
+    def __init__(self, *args, **kwargs):
+        super(CompositeData, self).__init__(self, *args, **kwargs)
+        self._children = []
```

Comment:
This should be instantiated in the constructor (e.g., through ``kwargs['children']``) and made immutable (e.g., a tuple rather than a list)